includes/WcfceMailProcessor.inc

```diff
@@ -0,0 +1,223 @@
+<?php
+
+/**
+ * @file
+ * Webform submission mail processor.
+ */
+
+class WcfceMailProcessor {
+
+  /**
+   * Email data (comes from hook_mail_alter()).
+   *
+   * @var array
+   */
+  private $message;
+
+  /**
+   * WcfceMailProcessor constructor.
+   *
+   * @param array $message
+   *   Email data (comes from hook_mail_alter()).
+   */
+  public function __construct(array $message) {
+    $this->message = $message;
+  }
+
+  /**
+   * File email as activity for all cases related to the webform.
+   *
+   * @return array|false
+   *   Entity data of the last activity added, empty array or FALSE
+   *   in case of error.
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  public function createActivity() {
+    if (!$this->isMessageValid()) {
+      return FALSE;
+    }
+
+    if (!civicrm_initialize()) {
+      return FALSE;
+    }
+
+    $assignee_id = $this->getContactIdByEmail($this->message['to']);
+    $creator_id  = $this->getContactIdByEmail($this->message['from']);
+
+    if (!$creator_id) {
+      return FALSE;
+    }
+
+    $eml = $this->composeEml();
+    $cases = &$this->message['params']['submission']->civicrm['case'];
+
+    // For all cases related to the webform:
+    foreach ($cases as $case) {
+      // Create activity.
+      $result = civicrm_api3('Activity', 'create', [
+        'sequential' => 1,
+        'activity_type_id' => 'Email',
+        'case_id' => $case['id'],
+        'subject' => $this->message['subject'],
+        'details' => trim($this->message['body'][0]),
+        'assignee_id' => $assignee_id,
+        'source_contact_id' => $creator_id,
+      ]);
+      // Attach eml file to it.
+      if (!empty($result['values'][0]['id'])) {
```

Comment:
We can just do 
```php
if (!empty($result['id']))
```

Reply:
But in the result you pasted

```php

[
    "is_error": 0,
    "version": 3,
    "count": 1,
    "id": 2,   //Here is result['id']
    "values": [
        "0": [
            "id": "2",
            "source_record_id": "",
            "activity_type_id": "3",
            ...
        ]
    ]
]

```

The ID can be picked from line 5 above ^^. That's what I meant